### PR TITLE
Added geometric distribution

### PR DESCRIPTION
**Pull request by dowi**
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
Added math functions for computing Geometric distribution's PMF and CDF
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java [hunk 1]
```diff
@@ -704,6 +705,32 @@ public static double betaCdf(
         return distribution.cumulativeProbability(value);
     }
 
+    @Description("geometric pmf P(X = k) given p probability for success")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double geometricPmf(
+            @SqlType(StandardTypes.DOUBLE) double p,
+            @SqlType(StandardTypes.INTEGER) double k)
```
Comment (leepface) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:713]: You've got a type mismatch here sql INTEGER and java double

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java [hunk 2]
```diff
@@ -704,6 +705,32 @@ public static double betaCdf(
...
+        GeometricDistribution distribution = new GeometricDistribution(p);
+        return distribution.probability(k);
+    }
+
+    @Description("geometric cdf P(X <= k), given p probability for success")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double geometricCdf(
+            @SqlType(StandardTypes.DOUBLE) double p,
+            @SqlType(StandardTypes.INTEGER) double k)
```
Comment (leepface) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java:726]: Same

**Comment by leepface** [comment]
Code review:
- [ ] squash commits together into one

**Comment by dowi** [comment]
Commits squashed

**Comment by talgalili** [comment]
Hey @dowi  - I see that this diff has merge conflicts, could you please resolve them?

**Pull request closed without merging**